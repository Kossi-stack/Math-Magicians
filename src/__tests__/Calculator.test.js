import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Test Calculator', () => {
  test('renders Calculator correctly', () => {
    const component = renderer.create(<Calculator />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Calculator', () => {
  test('Press one number response', () => {
    render(<Calculator />);
    const result = screen.getByRole('none');
    fireEvent.click(screen.getByText('1'));
    expect(result.innerHTML).toBe('1');
  });

  test('Press two numbers response', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('2'));
    const result = screen.getByRole('none');
    expect(result.innerHTML).toBe('12');
  });
});
