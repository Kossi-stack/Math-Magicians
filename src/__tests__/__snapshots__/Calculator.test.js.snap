// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator renders Calculator correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="output"
  >
    <p
      className="result"
      role="none"
    >
      0
    </p>
  </div>
  <div
    className="grid"
  >
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="AC"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="+/-"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="%"
    />
    <input
      className="button grid-item bg-orange"
      onClick={[Function]}
      type="button"
      value="รท"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="7"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="8"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="9"
    />
    <input
      className="button grid-item bg-orange"
      onClick={[Function]}
      type="button"
      value="x"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="4"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="5"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="6"
    />
    <input
      className="button grid-item bg-orange"
      onClick={[Function]}
      type="button"
      value="-"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="1"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="2"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="3"
    />
    <input
      className="button grid-item bg-orange"
      onClick={[Function]}
      type="button"
      value="+"
    />
    <input
      className="button grid-item span-2-col"
      onClick={[Function]}
      type="button"
      value="0"
    />
    <input
      className="button grid-item "
      onClick={[Function]}
      type="button"
      value="."
    />
    <input
      className="button grid-item bg-orange"
      onClick={[Function]}
      type="button"
      value="="
    />
  </div>
</div>
`;
